index.js

```diff
@@ -29,25 +32,55 @@ if (mainFilename === 'pm2_loader.js') {
 
 var hostname = os.hostname();
 
-var logger = new (winston.Logger)({
-  rewriters: [
-    function(label, msg, meta) {
-      meta = fclone(meta); // Remove circular references.
-      meta.mainFilename = mainFilename;
-      meta.hostname = hostname;
+var rewriters = [
+  function(label, msg, meta) {
+    meta = fclone(meta); // Remove circular references.
+    meta.mainFilename = mainFilename;
+    meta.hostname = hostname;
+    return meta;
+  }
+]
+
+if (isECS) {
+  rewriters.push(function(label,msg, meta) {
+    // First, check if the meta object is actually an error. In this case we want to preserve
+    // the error message (which would otherwise be clobbered by the log message). Winston populates
+    // the `name` field of the meta object with the class of the object if available, so just check
+    // for `name === 'Error'` here.
+    if (meta.name === 'Error') {
+      meta.error_message = meta.message
+      delete meta.message;
+      delete meta.name; // This field just looks confusing.
       return meta;
     }
-  ]
-});
 
-var prod = process.env.NODE_ENV === 'prod';
+    // Otherwise, we expect the meta object to be tags for the log line. Apply some rudimentary
+    // sanitizing here to filter out unexpected nested objects. All of our tags should be either
+    // string or number values.
+    for (var key of Object.keys(meta)) {
+      var value = meta[key];
+      var isString = typeof value === 'string';
+      var isNumber = typeof value === 'number';
+      if (!isString && !isNumber) {
+        meta[key] = 'Redacted due to non-string/number value';
```

Comment:
Is the intent for us to filter these out in Vector so that they do not clutter up the logging with empty calories?

Reply:
These previously caused issues with type conflicts and other weirdness in logz, figured it'd be simpler to just align on always using numbers/strings for fields.

Reply:
Oh, I agree with that; I should have been clearer -- is the intent to filter out the redacted fields downstream? Having a bunch of "fields" in our logs that all say "Redacted due to non-string/number value" doesn't seem helpful :)

Reply:
Oh sorry - I figured I'd leave them in, it would at least give people looking at the logs some indication of what happened to their missing fields, rather than just having them vanish without a trace. We can modify if it ends up spammy.